unittests/FEXLinuxTests/tests/vdso/vdso_test.cpp

```diff
@@ -0,0 +1,163 @@
+getres_type getres_vdso = (getres_type)::clock_getres;
+
+using getcpu_type = int (*) (uint32_t *cpu, uint32_t *node);
+getcpu_type getcpu_vdso = (getcpu_type)::getcpu;
+
+class VDSOParser {
+#if __SIZEOF_POINTER__ == 8
+  using ELFHeader = Elf64_Ehdr;
+  using ELFSectionHeader = Elf64_Shdr;
+  using ELFSymbol = Elf64_Sym;
+#else
+  using ELFHeader = Elf32_Ehdr;
+  using ELFSectionHeader = Elf32_Shdr;
+  using ELFSymbol = Elf32_Sym;
+
+#endif
+  public:
+    VDSOParser(uint8_t *Ptr) {
+      ELFHeader *Header = (ELFHeader *)Ptr;
+      uint64_t SectionHeaderOffset = Header->e_shoff;
+      uint16_t SectionHeaderCount = Header->e_shnum;
+      ELFSectionHeader *SHdrs = (ELFSectionHeader*)(&Ptr[SectionHeaderOffset]);
+
+      ELFSectionHeader const *DynamicSymHeader{nullptr};
+      ELFSectionHeader const *DynamicStringHeader{nullptr};
+      for (size_t i = 0; i < SectionHeaderCount; ++i) {
+        if (SHdrs[i].sh_type == SHT_STRTAB && SHdrs[i].sh_addr) {
+          DynamicStringHeader = &SHdrs[i];
+        }
+        if (SHdrs[i].sh_type == SHT_DYNSYM) {
+          DynamicSymHeader = &SHdrs[i];
+        }
+      }
+
+      size_t NumDynSymSymbols = DynamicSymHeader->sh_size / DynamicSymHeader->sh_entsize;
+      const char *DynStrTab = reinterpret_cast<const char*>(&Ptr[DynamicStringHeader->sh_offset]);
+
+      for (size_t i = 0; i < NumDynSymSymbols; ++i) {
+        uint64_t offset = DynamicSymHeader->sh_offset + i * DynamicSymHeader->sh_entsize;
+        ELFSymbol const *Symbol =
+          reinterpret_cast<ELFSymbol const *>(&Ptr[offset]);
+
+        char const * Name = &DynStrTab[Symbol->st_name];
+        if (Symbol->st_info != 0) {
+          uint8_t* SymbolPtr = Symbol->st_value + Ptr;
+          VDSOSymbols[Name] = SymbolPtr;
+          printf("Found VDSO symbol '%s' at %p\n", Name, SymbolPtr);
+        }
+
+      }
+    }
+
+    uint8_t *GetVDSOSymbol(const char *String) {
+      auto it = VDSOSymbols.find(String);
+      if (it != VDSOSymbols.end()) {
+        return it->second;
+      }
+
+      return nullptr;
+    }
+
+    std::map<std::string, uint8_t*> VDSOSymbols;
+};
+
+static void LoadVDSO() {
+  uint64_t Begin = ::getauxval(AT_SYSINFO_EHDR);
+  if (!Begin) {
+    printf("No VDSO\n");
+    return;
+  }
+
+  VDSOParser VDSO((uint8_t*)Begin);
+  auto it = VDSO.GetVDSOSymbol("__vdso_time");
+  if (it) {
+    time_vdso = reinterpret_cast<time_type>(it);
+  }
+
+  it = VDSO.GetVDSOSymbol("__vdso_gettimeofday");
+  if (it) {
+    gettimeofday_vdso= reinterpret_cast<gettimeofday_type>(it);
+  }
+
+  it = VDSO.GetVDSOSymbol("__vdso_clock_gettime");
+  if (it) {
+    gettime_vdso = reinterpret_cast<gettime_type>(it);
+  }
+
+  it = VDSO.GetVDSOSymbol("__vdso_clock_getres");
+  if (it) {
+    getres_vdso = reinterpret_cast<getres_type>(it);
+  }
+
+  it = VDSO.GetVDSOSymbol("__vdso_getcpu");
+  if (it) {
+    getcpu_vdso = reinterpret_cast<getcpu_type>(it);
+  }
+}
+
+
+int main() {
```

Comment:
I think this would benefit from being a Catch2 test, where we can test results more precisely than "did it crash or not?".

Not sure what effort is needed to do that with the current FexLinuxTests build setup though, so this is more of a wishlist item :)

Reply:
We need #1902 merged before we can easily import catch2, I think, so canoe works in the guest context

Reply:
Possibly, but the compiler would be setup as a chimera with mixed guest/host configurations, so likely we'll run into compilations.

Reply:
If we are going to implement catch2 in the FEXLinuxTests then it should be an indepedent task from this PR. No need to complicate this PR with changing test behaviours here.

Reply:
Sounds fair, though similar line of reasoning was already applied when FLT was originally introduced. Now is a good time to revisit this though, so I went and created #1972 for tracking this.